src/Data/LLVM/BitCode/IR/Globals.hs

```diff
@@ -50,18 +50,37 @@ parseGlobalVar n r = label "GLOBALVAR" $ do
   vis <- if length (recordFields r) > (6 + offset) && not (link `elem` [Internal, Private])
                 then field 6 visibility
                 else pure DefaultVisibility
+  tl <- if length (recordFields r) > (7 + offset) -- && not (link `elem` [Internal, Private])
+                then field 7 threadLocal
+                else pure NotThreadLocal
+
+  unnamed <-
+    if length (recordFields r) > (8 + offset)
+    then do
+      field 8 unnamedAddr
+    else return Nothing
+
+  addrspace <- if explicitTy
+                  then return . AddrSpace . fromIntegral $ shiftR mask 2
+                  else case ptrty of
+                         PtrTo as _ -> return as
+                         PtrOpaque as -> return as
+                         _ -> fail $ "Invalid type for value: " ++ show ptrty
```

Comment:
Can this `case` expression be simplified by defining it in terms of `ptrAddrSpace` from https://github.com/GaloisInc/llvm-pretty/pull/148?